src/config/ServerConfig.cpp

```diff
@@ -106,6 +117,18 @@ void ServerConfig::addLocation(const LocationConfig& location)
   _locations.push_back(location);
 }
 
+void ServerConfig::checkPortDuplicate(int port)
+{
+  for (std::vector<int>::const_iterator it = _ports.begin(); it != _ports.end();
+       ++it) {
+    if (port == *it) {
+      std::ostringstream oss;
+      oss << port;
+      throw std::invalid_argument("duplicated port " + oss.str());
+    }
+  }
```

Comment:
Just want to mention that these functions exist :D
```suggestion
  if (ft::contains(_ports, port)) {
    throw std::invalid_argument("duplicate port " + ft::to_string(port));
  }
```